mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRErrorResponse.java

```diff
@@ -0,0 +1,237 @@
+import javax.xml.bind.annotation.XmlRootElement;
+import javax.xml.bind.annotation.XmlTransient;
+import javax.xml.bind.annotation.adapters.XmlJavaTypeAdapter;
+
+import org.apache.logging.log4j.LogManager;
+import org.mycore.restapi.converter.MCRInstantXMLAdapter;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+
+@XmlRootElement(name = "error")
+@XmlAccessorType(XmlAccessType.PUBLIC_MEMBER)
+@JsonAutoDetect(getterVisibility = JsonAutoDetect.Visibility.PUBLIC_ONLY)
+public class MCRErrorResponse {
+
+    UUID uuid;
+
+    Instant timestamp;
+
+    String errorCode;
+
+    Throwable cause;
+
+    String message;
+
+    String detail;
+
+    int status;
+
+    public static MCRErrorResponse fromStatus(int status) {
+        final MCRErrorResponse response = new MCRErrorResponse();
+        response.setStatus(status);
+        return response;
+    }
+
+    private MCRErrorResponse() {
+        uuid = UUID.randomUUID();
+        timestamp = Instant.now();
+    }
+
+    public WebApplicationException toException() {
+        WebApplicationException e;
+        Response.Status s = Response.Status.fromStatusCode(status);
+        final Response response = Response.status(s)
+            .entity(this)
+            .build();
+        //s maybe null
+        switch (Response.Status.Family.familyOf(status)) {
+            case CLIENT_ERROR:
+                //Response.Status.OK is to trigger "default" case
+                switch (s != null ? s : Response.Status.OK) {
+                    case BAD_REQUEST:
+                        e = new BadRequestException(getMessage(), response, getCause());
+                        break;
+                    case FORBIDDEN:
+                        e = new ForbiddenException(getMessage(), response, getCause());
+                        break;
+                    case NOT_ACCEPTABLE:
+                        e = new NotAcceptableException(getMessage(), response, getCause());
+                        break;
+                    case METHOD_NOT_ALLOWED:
+                        e = new NotAllowedException(getMessage(), response, getCause());
+                        break;
+                    case UNAUTHORIZED:
+                        e = new NotAuthorizedException(getMessage(), response, getCause());
+                        break;
+                    case NOT_FOUND:
+                        e = new NotFoundException(getMessage(), response, getCause());
+                        break;
+                    case UNSUPPORTED_MEDIA_TYPE:
+                        e = new NotSupportedException(getMessage(), response, getCause());
+                        break;
+                    default:
+                        e = new ClientErrorException(getMessage(), response, getCause());
+                }
+                break;
+            case SERVER_ERROR:
+                //Response.Status.OK is to trigger "default" case
+                switch (s != null ? s : Response.Status.OK) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A switch statement does not contain a break](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5532334)